src/lib.rs

```diff
@@ -335,13 +335,21 @@ impl Lzxd {
         // to return a continous slice. if we're called on non-aligned, we could shift things
         // and align it.
 
-        // FIXME: Why is the last block's size observed to be one?
-        if self.current_block.size > 1 {
-            // Align the window up to 32KB.
-            // See https://github.com/Lonami/lzxd/issues/7 for details.
-            if let Some(len) = 0x8000usize.checked_sub(decoded_len) {
-                self.window.zero_extend(len);
-                decoded_len += len;
+        // Align the window up to 32KB.
+        // See https://github.com/Lonami/lzxd/issues/7 for details.
+        if let Some(pad_len) = 0x8000usize.checked_sub(decoded_len) {
+            if pad_len > 0 {
+                // At this point, we know the output block is not aligned to 32KB. We'll subtract
+                // the padding bytes from the input block's size (as they are implicitly read from the block)
+                // and then zero-extend the window.
+                if let Some(len) = self.current_block.size.checked_sub(pad_len as u32) {
```

Comment:
This might be wrong for the last chunk - we might want to extend `self.current_block.size` bytes of padding instead.